lib/devicetrust/authz/authz.go

```diff
@@ -104,3 +104,24 @@ func verifyDeviceExtensions(dt *types.DeviceTrust, username string, verified boo
 		return nil
 	}
 }
+
+// CalculateTrustedDeviceRequirement calculates the requirement based on
+// the cluster config and user roles.
+func CalculateTrustedDeviceRequirement(
+	clusterDeviceTrust *types.DeviceTrust,
+	getRoles func() []types.Role,
+) types.TrustedDeviceRequirement {
+	// Required by cluster mode?
+	if dtconfig.GetEnforcementMode(clusterDeviceTrust) == constants.DeviceTrustModeRequired {
```

Comment:
Previously, `dtconfig.GetEffectiveMode()` was used, which also checks if the module is not OSS. Since this doesn't work client-side, I replaced it with `dtconfig.GetEnforcementMode()`. This change should have no impact on the server-side, as it already ensures the build is enterprise.